README.md

```diff
@@ -316,6 +316,21 @@ record :with_date_array
 end
 ```
 
+### User-defined Types
```

Comment:
I agree that the terminology is tricky here. I thought about using `type_alias` so maybe I'll change it to that.